Codechecker fixes

lucaboesch opened the pull request:
Adhere to more recent Moodle Coding Guidelines.

lucaboesch committed:
Short array syntax.

lucaboesch committed:
Some fixes to make codechecker happy.

danmarsden commented on classes/import/marksessions.php:
```diff
@@ -132,15 +132,15 @@ protected function get_column_data($row, $index) {
      * @param bool $useprogressbar Whether progress bar should be displayed, to avoid html output on CLI.
      */
-    public function __construct($text = null, $att, $encoding = null, $delimiter = null, $importid = 0,
+    public function __construct($att, $text = null, $encoding = null, $delimiter = null, $importid = 0,
```
did you change the order of these params on purpose? - this doen'st seem to be a codechecker change?

danmarsden commented on import/marksessions.php:
```diff
@@ -66,12 +66,12 @@
 if (optional_param('needsconfirm', 0, PARAM_BOOL)) {
     $form = new \mod_attendance\form\import\marksessions($url->out(false), $formparams);
 } else if (optional_param('confirm', 0, PARAM_BOOL)) {
-    $importer = new \mod_attendance\import\marksessions(null, $att, null, null, $importid);
```
looks like here'e the matching change for the function parameter order change - I think we should probably do this as it's own commit rather than bundling into the "codechecker fixes" if there's a good reason we should do this.

lucaboesch committed:
Invert __construct function parameters of marksessions

Realized the order of the paramaters so that no mandatory
follows an optional one.

lucaboesch commented:
Well spotted, @danmarsden.
I've added an individual commit for it.
Note that the catalyst automation seems still using actions/checkout@v3 instead of actions/checkout@v4.

Best,
Luca

danmarsden commented:
cool thanks! - merged it in.
